### PR TITLE
Getting and deleting deployments and labels, Getting metrics

### PR DESCRIPTION
**Pull request by siredmar**
Added listing and deleting deployments
Added labels (tags), type (cloud/edge) to give more control where to deploy deployments
Added metrics to get information about the deployments states.

### CODE REVIEW

File: service/internal/deployment/manifest/manifest_test.go [hunk 1]
```diff
@@ -13,6 +13,7 @@ func TestCreateLayeredManifest(t *testing.T) {
 
 	c := CustomerManifest{
 		Application: "myapplication",
+		Type:        "edge",
```
Comment (KairaMiu) [inline service/internal/deployment/manifest/manifest_test.go:16]: Please add a check it this is considered the correct way -> Tag with `type: edge`

**Review by KairaMiu — CHANGES_REQUESTED** [review]
Only one minor and one editorial open.

File: service/internal/deployment/manifest/manifest_test.go (hunk 1 above)
Comment (siredmar) [reply service/internal/deployment/manifest/manifest_test.go:16]: No way to check this as this is not handeled in the IoT Hub manifest file...

File: service/api/api_deployments_delete.go [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+	"log"
+	"net/http"
+	"strings"
+
+	"github.com/edgefarm/edgefarm.alm/internal/deployment"
+	iothub "github.com/edgefarm/edgefarm.alm/internal/iothubwrapper"
+	token "github.com/edgefarm/edgefarm.alm/internal/token"
+)
+
+// DeploymentsDeletePut applies a customer manifest to the tenants devices
```
[the remark this answers is not in the document]
Comment (siredmar) [reply service/api/api_deployments_delete.go]: fixed.

**Review by KairaMiu — APPROVED** [review]

**Pull request merged**